week1/homework/src/server.js

```diff
@@ -2,14 +2,43 @@
 
 const http = require('http');
 
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
- */
+
 function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+
+    function stateFunction(response, state) {
+
+      response.writeHead(200, { 'Content-Type': 'application/json' });
+      response.write(JSON.stringify({ state }));;
+      response.end();
+    }
+
+
+    if (request.url === "/state") {
+      stateFunction(response, state);
+
+    } else if (request.url === '/add') {
+      state += 1;
+      stateFunction(response, state);
+    } else if (request.url === '/subtract') {
+      state -= 1;
+      stateFunction(response, state);
+    }
+    else if (request.url === "/reset") {
+
+      state = 10;
+      stateFunction(response, state);
+
```

Comment:
Try to have one style of coding you're now mixing the following:
```
if (condition) {
    code;

}
```
```
} else if (condition) {
    code;
}
```
```
} 
else if (condition) {
    code;
}
```
```
} else if (condition) {

    code;

}
```
They are all okay on their own, and it will work when you mix these styles, but it's more readable if you pick just one.
You should choose one style for newlines and formatting and stick with that.